### DISCUSSION TITLE
Prevent the node border from interfering with the node background

### DISCUSSION THREAD

**Discussion by aaa1386** [freeplane/freeplane#1514] [Ideas]
With the increase in thickness, the border node should go outside the node to become independent from the background instead of going inside the node.

https://github.com/freeplane/freeplane/assets/38586259/ebd812c5-5286-4c84-b3fa-825094932583

If only the border node is set for the selected node, in some cases, the background of the node that is clicked will interfere with the settings of the selected node.

https://github.com/freeplane/freeplane/assets/38586259/d92cdd03-177a-427c-b7ed-0bc79208b5da

In the figure below, the order to increase the thickness of the border node is shown with an arrow in the figure below. This is the solution to the problem.

![image](https://github.com/freeplane/freeplane/assets/38586259/b188caf2-d72b-4595-9a10-8e1eb89f8ac5)

**Comment by aaa1386** [comment]
If the software can work like the figure I have drawn, many good things will happen.

![image](https://github.com/freeplane/freeplane/assets/38586259/f488529b-d799-48e4-9959-3e760766da80)

**Comment by aaa1386** [comment]
In the figure above, the node border is separated from the background node and they are completely independent from each other

**Comment by aaa1386** [comment]
This is how Freeplane currently works:
![image](https://github.com/freeplane/freeplane/assets/38586259/75feedff-6fea-4134-9948-5fbb9c0f11cc)


Suggested model to the software
![image](https://github.com/freeplane/freeplane/assets/38586259/a601ee6f-94b8-4e18-baed-810fa58ab820)

**Comment by euu2021** [comment]
> If the software can work like the figure I have drawn, many good things will happen.

It's still not clear to me what are the advantages and disadvantages of each approach.

**Comment by dpolivaev** [comment]
I see big potential of the approach. 
<img width="981" alt="thickBorders" src="https://github.com/freeplane/freeplane/assets/1558052/834c1fe8-12ee-4319-bdf0-ded8e614cf47">

And I have just implemented it, so you can try it out in the last preview 1.11.8_08.

[thickBorders.mm.zip](https://github.com/freeplane/freeplane/files/13244207/thickBorders.mm.zip)

**Reply by dpolivaev** [reply]
I think it can be the new representation, not a configurable option.

**Reply by i-plasm** [reply]
@dpolivaev looks nice! I wonder, though, if it would interfere too much with clone marks? I already have trouble visualizing them in some of my maps. With outer border I wonder how clone marks would be dealt with? Maybe outer/inner should be a configuration in preferences, unless there's a better way to make nodes that are clones to be marked and evident somehow. Thanks!

**Reply by dpolivaev** [reply]
I checked it, it works fine with clone marks too.
<img width="834" alt="cloneMarks" src="https://github.com/freeplane/freeplane/assets/1558052/0264c053-384f-4d07-9053-db4c5f26c840">

**Reply by i-plasm** [reply]
Good to know. It would be great if clone marks had a preference that allows the user to select its size. They are often really tiny in my maps. I'd love if they could stand out more.

**Reply by euu2021** [reply]
> Good to know. It would be great if clone marks had a preference that allows the user to select its size. They are often really tiny in my maps. I'd love if they could stand out more.

It exists, but the name is not very intuitive: `Preferences…->Appearance->Size limits->State symbol width`

**Reply by i-plasm** [reply]
Thank you! It will help me

**Reply by dpolivaev** [reply]
Can you suggest a better name? For example, what do you think about "Node state indicator size"?

**Reply by i-plasm** [reply]
> Can you suggest a better name? For example, what do you think about "Node state indicator size"?

I think a name could be something like "Cloned node state indicator size" (that is, assuming the word "state" is exclusively talking about cloning, not some other state...)

**Reply by dpolivaev** [reply]
Actually, the same option changes also indicators for minimized nodes, but no others.

**Reply by euu2021** [reply]
I suggest including "clones" and "minimized" in the name, thus making it easy to be discovered when searching in the Command Search.

So, the name could be something like "State symbol width (clone symbol; minimized symbol)"
